plugin/pkg/proxy/connect.go

```diff
@@ -142,6 +143,10 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 
 				// Clear AD bit in case request had set the AD bit. The empty response is not authenticated.
 				newRet.AuthenticatedData = false
+
+				// Clear AA bit in case request had set the AA bit.
```

Comment:
Why do you clear the AA bit?

Reply:
Cool, no concerns. Please add that description to the existing comment.

Reply:
>  it's good to keep it set to false

Actually, are you basing that on common sense, or an RFC requirement?
We may want to follow what is done for actual truncated responses.  

If RFC says AA should be set to false when replying with a truncated response, then we should definitely do that here.
if not, then we need to discuss pros/cons.  E.g. what's is the negative of leaving AA bit as it was.

Reply:
I couldn't find any specific RFC that directly addresses the handling of the authorization bit (AA) in truncated DNS responses. There is no specific handling or interpretation defined for the authorization bit (AA) in truncated responses

RFC 1035 does not specifically cover the handling of the AA bit in truncated responses
https://tools.ietf.org/html/rfc1035
https://datatracker.ietf.org/doc/html/rfc1035#section-4.2.1

When a response is truncated, the focus is on addressing the truncation itself and ensuring the complete response is obtained. The authorization bit is not directly related to handling truncation but rather serves as an indicator of authority.
The authorization bit is considered meaningful only in non-truncated responses, where it signifies the authoritative nature of the responding server.

AFAIK - It is better to keep it set to false here. I don't know if there is any negative of leaving AA bit as it was. I will go with what you suggest.